bundle/src/deepracer_simulation_environment/scripts/mp4_saving/training_image_editing.py

```diff
@@ -58,6 +62,69 @@ def _edit_major_cv_image(self, major_cv_image, cur_training_phase):
+            
+            # Last lap time
+            last_lap_time = mp4_video_metrics_info[self.racecar_index].last_lap_time
+            # The initial default last_lap_time from s3_metrics.py is inf
+            # If the ros service in s3_metrics.py has not come up yet, last_lap_time is 0
+            last_lap_time = utils.milliseconds_to_timeformat(
+                datetime.timedelta(milliseconds=last_lap_time)) \
+                if last_lap_time != float("inf") and last_lap_time != 0 else "--:--.---"    
+            last_lap_time_text = "Last lap | {}".format(last_lap_time) 
+            major_cv_image = utils.write_text_on_image(image=major_cv_image, text=last_lap_time_text,
+                                                    loc=(width-180, loc_y+25), font=self.amazon_ember_light_18px,
+                                                    font_color=RaceCarColorToRGB.White.value,
+                                                    font_shadow_color=RaceCarColorToRGB.Black.value)
+            
+            # Progress
+            progress_text = "Progress | {}%".format(round(mp4_video_metrics_info[self.racecar_index].completion_percentage,2)) 
+            major_cv_image = utils.write_text_on_image(image=major_cv_image, text=progress_text,
+                                                    loc=(width-180, loc_y+50), font=self.amazon_ember_light_18px,
+                                                    font_color=RaceCarColorToRGB.White.value,
+                                                    font_shadow_color=RaceCarColorToRGB.Black.value)
+            
+            # Steering Angle
+            steering_text = "Steering | {}".format(mp4_video_metrics_info[self.racecar_index].steering)
+            major_cv_image = utils.write_text_on_image(image=major_cv_image, text=steering_text,
+                                                    loc=(loc_x, loc_y), font=self.amazon_ember_light_18px,
+                                                    font_color=RaceCarColorToRGB.White.value,
+                                                    font_shadow_color=RaceCarColorToRGB.Black.value)
+            
+            # Throttle
+            loc_y += 25
+            steering_text = "Throttle | {}".format(round(mp4_video_metrics_info[self.racecar_index].throttle,3))
+            major_cv_image = utils.write_text_on_image(image=major_cv_image, text=steering_text,
+                                                    loc=(loc_x, loc_y), font=self.amazon_ember_light_18px,
+                                                    font_color=RaceCarColorToRGB.White.value,
+                                                    font_shadow_color=RaceCarColorToRGB.Black.value)
+            
+            # Speed
+            loc_y += 25
+            speed_text = "Speed | {} m/s".format(utils.get_speed_formatted_str(mp4_video_metrics_info[self.racecar_index].speed))
```

Comment:
Is formatting needed? Or is the input already rounded?

Reply:
The input is already rounded by utils.get_speed_formatted_str()

This is the way it's already displayed in existing overlays for speed.